### PR TITLE
[Techdocs]Convert `sanitizeDOM` to hook 

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

> **Note:** all changes were pushed in small commits to help with the review process.

Convert `sanitizeDOM` to hook as part of code readability improvements in the dom file: https://github.com/backstage/backstage/pull/11633.

File: `plugins/techdocs/src/reader/components/TechDocsReaderPageContent/dom.tsx`.

- Extracts the sanitizer hooks to separate files;
- Creates a hook that reads sanitizer configuration;
- And creates another hook that returns a transformer responsible for sanitizing a given dom element.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs/src/reader/transformers/html/hooks/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export { removeUnsafeLinks } from './links';
+export { removeUnsafeIframes } from './iframes';
```
Comment (realandersn) [inline plugins/techdocs/src/reader/transformers/html/hooks/hooks.ts:18]: Why is this file needed? Could be exported from `index.ts`?

File: plugins/techdocs/src/reader/transformers/html/hooks/iframes.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (realandersn) [inline plugins/techdocs/src/reader/transformers/html/hooks/iframes.ts:1]: maybe rename file `removeUnsafeIframes`, could be clearer?

File: plugins/techdocs/src/reader/transformers/html/hooks/links.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (realandersn) [inline plugins/techdocs/src/reader/transformers/html/hooks/links.ts:1]: maybe rename file `removeUnsafeLinks`, could be clearer?

File: plugins/techdocs/src/reader/transformers/html/transformer.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  return useMemo(() => {
+    return configApi.getOptionalConfig('techdocs.sanitizer');
+  }, [configApi]);
+};
+
+/**
+ * Returns a transformer that sanitizes the dom's internal html.
+ */
+export const useHtmlTransformer = (): Transformer => {
```
Comment (realandersn) [inline plugins/techdocs/src/reader/transformers/html/transformer.ts:39]: Optional: Maybe consider the naming here. I think it was a bit clearer to have sanitize in the name somewhere? But maybe this should do more in the future?

**Review by realandersn — COMMENTED** [review]
Just some minor comments.

File: plugins/techdocs/src/reader/transformers/html/hooks/iframes.ts (hunk 2 above)
Comment (camilaibs) [reply plugins/techdocs/src/reader/transformers/html/hooks/iframes.ts:1]: Let me give you some context, I used _iframe_ in the filename because this file contains everything related to iframes, including the `removeUnsafeIframes` function, and others: `isSafe` and `isIframe`.
Does it make sense to you?

File: plugins/techdocs/src/reader/transformers/html/hooks/hooks.ts (hunk 1 above)
Comment (camilaibs) [reply plugins/techdocs/src/reader/transformers/html/hooks/hooks.ts:18]: I started everything in a single file and then extract it into other separate ones, it ended up only with the import lines, I will remove it, thanks for the good eyes on it!

File: plugins/techdocs/src/reader/transformers/html/transformer.ts (hunk 4 above)
Comment (camilaibs) [reply plugins/techdocs/src/reader/transformers/html/transformer.ts:39]: Cool, will do!

File: plugins/techdocs/src/reader/transformers/html/hooks/links.ts (hunk 3 above)
Comment (camilaibs) [reply plugins/techdocs/src/reader/transformers/html/hooks/links.ts:1]: Same as [here](https://github.com/backstage/backstage/pull/11647/files#r879283785), what do you think?

File: plugins/techdocs/src/reader/transformers/html/hooks/iframes.ts (hunk 2 above)
Comment (realandersn) [reply plugins/techdocs/src/reader/transformers/html/hooks/iframes.ts:1]: Yes, makes sense!

File: plugins/techdocs/src/reader/transformers/html/transformer.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import DOMPurify from 'dompurify';
+import { useMemo, useCallback } from 'react';
+
+import { useApi, configApiRef } from '@backstage/core-plugin-api';
+
+import { Transformer } from '..';
```
Comment (realandersn) [inline plugins/techdocs/src/reader/transformers/html/transformer.ts:22]: importing from '..' is frowned upon I've learnt 😅 should probably be '../transformer' instead.

**Review by realandersn — APPROVED** [review]
Approving! Just left one more small comment.

File: plugins/techdocs/src/reader/transformers/html/transformer.ts (hunk 5 above)
Comment (camilaibs) [reply plugins/techdocs/src/reader/transformers/html/transformer.ts:22]: TIL
